### ISSUE TITLE
SwampDragon + Auth + NGINX => Session Problem

### ISSUE THREAD

**Issue by ZuSe** [jonashagstedt/swampdragon-auth#12]
Hi folks,

I have set up an environment related to this guide: http://swampdragon.net/blog/deploying-swampdragon/

I use dev1.mydomain.com (with SSL) for django and dev1sd.mydomain.com (also with SSL) for swampdragon. 

swampdragon-auth is installed and highly needed. Connecting to the Websocket with the JavaScript-Client works fine, but when I try to subscribe to a channel I get {signin_required: "you need to sign in"}
returned. 
As far as I understand this behaviour is caused by my subdomain setup. I have set 
SESSION_COOKIE_DOMAIN = ".mydomain.com" in settings.py, but I still get the error.

On Localhost everything works fine.

I would really appreciate any advice that might help me to solve this issue :)

**Comment by jonashagstedt** [comment]
Are you using the right connection class?

`SWAMP_DRAGON_CONNECTION = ('swampdragon_notifications.notification_connection.Connection', '/data')`

Also if you put a breakpoint anywhere in your router can you inspect the `self.connection.user` property?

**Comment by ZuSe** [comment]
Nope, i use 
# SwampDragon settings
SWAMP_DRAGON_CONNECTION = ('swampdragon_auth.socketconnection.HttpDataConnection', '/data') 

what's the difference to your connection class?

Setting a breakpoint is a bit hard since it's deployed to a remote system. Need to setup a local copy of it first, will post you the object when it's done. Anyway, thanks for your help so far :)

**Comment by jonashagstedt** [comment]
Sorry I'm being a lemon, I copied the SWAMPDRAGON_CONNECTION from the wrong project, you are using the right one

**Comment by jonashagstedt** [comment]
You can set a break point on the server, you can just run the projects on different ports and ssh in 

i.e 
`./manage.py runsd localhost:9988`

**Comment by ZuSe** [comment]
Hi jonas, it works now with the settings described in my first post.
Seems like the problem was related to some caching stuff... i have just cleared everything I found.

Sorry for your time and thanks again for your help.

**Comment by jonashagstedt** [comment]
No worries.

If someone else comes across this issue, how would you recommend them to proceed?
What should they check and be aware of?

Thanks for the feedback

**Comment by ZuSe** [comment]
Hi jonas,

here what I did: 
1. Check if the proper SESSION_COOKIE_DOMAIN is set in settings.py:
It has to look like SESSION_COOKIE_DOMAIN = ".mydomain.com", not "xxx.mydomain.com"

2. Clear your cookies for the TLD and subdomains in your browser
3. Remove all pycache files (I did it, dunno if necessary)
4. Restart the uwsgi service
5. Restart swampdragon (usually via supervisord in a productive env)
6. Restart nginx (maybe not needed but I did to clear the cache)

Usually you don't see the error message mentioned in my first post but only "Cannot read length of undefined" in the JavaScript console (thrown at setupChannel). If this happens print the object to get the message.

PS: I have another question, but that might be better discussed in it's own issue.
